spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/it/DatastoreIntegrationTests.java

```diff
@@ -739,6 +742,45 @@ void lazyReferenceTest() throws InterruptedException {
     assertThat(loadedParent).isEqualTo(lazyParentEntity);
   }
 
+  @Test
+  void lazyReferenceUpdateAndSaveParentTest() {
+    LazyEntity lazyParentEntity = new LazyEntity(new LazyEntity(new LazyEntity()));
+    this.datastoreTemplate.save(lazyParentEntity);
+
+    LazyEntity loadedParent =
+        this.datastoreTemplate.findById(lazyParentEntity.id, LazyEntity.class);
+
+    // update lazyReferenced property
+    LazyEntity newChild = new LazyEntity();
+    loadedParent.setLazyChild(newChild);
+    this.datastoreTemplate.save(loadedParent);
+
+    loadedParent = this.datastoreTemplate.findById(loadedParent.id, LazyEntity.class);
+    assertThat(loadedParent.getLazyChild()).isEqualTo(newChild);
```

Comment:
`LazyChild` has `Long` type id, so before saving to datastore, it will get auto-allocated to a unique key when it's not set ([here](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/c5bbec361900e3806f71c360cdf0de8cc8e9ecf5/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/DatastoreTemplate.java), [allocation logic](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/e04f755f75c98619c11fde42106cff81e95f1540/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/DatastoreServiceObjectToKeyFactory.java#L86)).